src/main/java/spoon/support/compiler/VirtualFolder.java

```diff
@@ -46,10 +46,10 @@ public void addFolder(SpoonFolder o) {
 	@Override
 	public List<SpoonFile> getAllFiles() {
 		List<SpoonFile> result = new ArrayList<>();
-
+		Set<SpoonFile> distinctSet = new HashSet<>();
 		for (SpoonFile f : getFiles()) {
 			// we take care not to add a file that was already found in a folder
-			if (!result.contains(f)) {

```

Comment:
The not using HashSet means a performance problems with bigger number of files. The MS Windows file.equals() method is much faster then File.getCannonicalFile, but it is still quite slow. So recursive calling of it in List.contains needs time too. I will measure it tomorrow and give you exact number.